[GTK4] Scrolling with touchpad is broken on Wayland

alice-mkh opened the pull request:
#### 9b9c04184c998ba6efdd8931324097bb5e2489bc
<pre>
[GTK4] Scrolling with touchpad is broken on Wayland
<a href="https://bugs.webkit.org/show_bug.cgi?id=238203">https://bugs.webkit.org/show_bug.cgi?id=238203</a>

GTK 4.7.x has new API to know whether scroll deltas are continuous or
discrete, and the former don&apos;t have a multiplier anymore. Adjust event
handling accordingly.

Reviewed by Michael Catanzaro.

* Source/WebKit/Shared/gtk/WebEventFactory.cpp:
(WebKit::WebEventFactory::createWebWheelEvent):
* Source/WebKit/UIProcess/gtk/ViewGestureControllerGtk.cpp

Canonical link: <a href="https://commits.webkit.org/253069@main">https://commits.webkit.org/253069@main</a>
</pre>

mcatanzaro commented on Source/WebKit/UIProcess/gtk/ViewGestureControllerGtk.cpp:
```diff
@@ -39,7 +39,11 @@ static const Seconds swipeMaxAnimationDuration = 400_ms;
 static const double swipeAnimationBaseVelocity = 0.002;
 
 // GTK divides all scroll deltas by 10, compensate for that
```
This comment belongs inside the `#else` now, right?

alice-mkh replied:
Yep.

alice-mkh committed:
[GTK4] Scrolling with touchpad is broken on Wayland
https://bugs.webkit.org/show_bug.cgi?id=238203

GTK 4.7.x has new API to know whether scroll deltas are continuous or
discrete, and the former don't have a multiplier anymore. Adjust event
handling accordingly.

Reviewed by Michael Catanzaro.

* Source/WebKit/Shared/gtk/WebEventFactory.cpp:
(WebKit::WebEventFactory::createWebWheelEvent):
* Source/WebKit/UIProcess/gtk/ViewGestureControllerGtk.cpp

Canonical link: https://commits.webkit.org/253069@main
